delfin/drivers/pure/flasharray/pure_flasharray.py

```diff
@@ -363,6 +364,134 @@ def reset_connection(self, context, **kwargs):
     def get_access_url():
         return 'https://{ip}'
 
+    def collect_perf_metrics(self, context, storage_id, resource_metrics,
+                             start_time, end_time):
+        metrics = []
+        if resource_metrics.get(constants.ResourceType.STORAGE):
+            arrays_id, arrays_name = self.get_array()
+            storage_metrics = self.get_metrics(
+                storage_id,
+                resource_metrics.get(constants.ResourceType.STORAGE),
+                start_time, end_time,
+                self.rest_handler.REST_ARRAY_HISTORICAL_URL,
+                constants.ResourceType.STORAGE, arrays_id, arrays_name)
+            metrics.extend(storage_metrics)
+        if resource_metrics.get(constants.ResourceType.VOLUME):
+            storage_metrics = self.get_metrics(
+                storage_id,
+                resource_metrics.get(constants.ResourceType.VOLUME),
+                start_time, end_time,
+                self.rest_handler.REST_VOLUME_HISTORICAL_URL,
+                constants.ResourceType.VOLUME)
+            metrics.extend(storage_metrics)
+        return metrics
+
+    def get_metrics(self, storage_id, resource_metrics, start_time, end_time,
+                    url, resource_type, resource_id=None, resource_name=None):
+        metrics = []
+        local_time = time.mktime(time.localtime())
+        localtime_ms = local_time * units.k
+        if start_time < localtime_ms - consts.ONE_YEAR_DIFFERENCE\
+                or end_time < start_time or start_time > localtime_ms:
+            return metrics
+        historical = self.get_historical(localtime_ms, start_time)
+        list_metrics = self.rest_handler.rest_call(url.format(historical))
+        for storage_metrics in (list_metrics or []):
+            opened = storage_metrics.get('time')
+            time_difference = local_time - time.mktime(time.gmtime())
+            timestamp = (int(datetime.datetime.strptime
+                             (opened, '%Y-%m-%dT%H:%M:%SZ').timestamp()
```

Comment:
It is recommended to set `%Y-%m-%dT%H:%M:%SZ` as a constant in `consts.py` .

Reply:
Has been modified